service/dap/server.go

```diff
@@ -469,6 +469,123 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
+	if mode == "record" {
+		s.config.Debugger.Backend = "rr"
+		s.send(&dap.CapabilitiesEvent{ Event: *newEvent("capabilities"), Body: dap.CapabilitiesEventBody{Capabilities: dap.Capabilities{ SupportsStepBack: true }}})
+
+		// Validate rr availability
+		if err := gdbserial.CheckRRAvailable(); err != nil {
+			s.sendErrorResponseWithURL(request.Request, FailedToInitialize, "Unable to find  rr in record configuration", err.Error(), "https://github.com/rr-debugger/rr/wiki/Building-And-Installing", "rr installation page")
+			return
+		}
+
+		// Record only works on rr trace directories
+		traceDirectory, ok := request.Arguments["traceDirectory"].(string)
+		if !ok || traceDirectory == "" {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The traceDirectory attribute is missing in record configuration.")
+			return
+		}
+
+		buildFlags := ""
```

Comment:
I believe this code is duplicated from `mode == "debug"`, if `mode == "record"` is needed for backwards compatibility then it's fine to have it, but it should just be converted into `mode = "debug"` internally, with the backend set to `rr`.